gradio/blocks.py

```diff
@@ -896,7 +896,10 @@ def get_layout(block: Block):
         config["layout"] = get_layout(root_block)
 
         config["components"] = []
-        for _id, block in self.blocks.items():
+        blocks_items = list(
+            self.blocks.items()
+        )  # freeze as list to prevent concurrent re-renders from changing the dict during loop
+        for _id, block in blocks_items:
```

Comment:
Should we actually just put a lock around `self.blocks.items()` so that only one re-render can affect the list at a time? I'm wondering if there's a scenario where two re-renders happen and the second one "overrides" the first one

Reply:
That is not necessary, and locks could lead to blocking code. This should be sufficient to prevent any concurrent writes and read.

Reply:
Sg should be good to merge then